iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputFormat.java

```diff
@@ -83,9 +83,19 @@ private static HiveIcebergRecordWriter writer(JobConf jc) {
         .operationId(operationId)
         .build();
     String tableName = jc.get(Catalogs.NAME);
-    HiveFileWriterFactory hfwf = new HiveFileWriterFactory(table, fileFormat, schema,
-        null, fileFormat, null, null, null, null);
-    return new HiveIcebergRecordWriter(schema, spec, fileFormat,
-        hfwf, outputFileFactory, io, targetFileSize, taskAttemptID, tableName);
+    if (HiveIcebergStorageHandler.isDelete(jc, tableName)) {
+      // TODO: remove this Avro-specific logic once we have Avro writer function ready
+      // for now, this means that Avro delete files will not contain the 'row' column
+      Schema positionDeleteRowSchema = fileFormat == FileFormat.AVRO ? null : schema;
+      HiveFileWriterFactory hfwf = new HiveFileWriterFactory(table, fileFormat, schema,
+          null, fileFormat, null, null, null, positionDeleteRowSchema);
+      return new HiveIcebergDeleteWriter(hfwf, schema, spec, fileFormat, outputFileFactory, io, targetFileSize,
+          taskAttemptID, tableName);
```

Comment:
Why do we need different `HiveFileWriterFactory` for different cases? I thought that the factory hides that abstraction

Reply:
hmm, I did it this way mostly because the hfwf in the delete case needs the additional `positionDeleteRowSchema ` populated. However, I guess we could supply the delete row schema for the simple write path too, it just wouldn't be used. I'll give it a try